src/goto-instrument/contracts/dynamic-frames/dfcc_instrument.h

```diff
@@ -12,15 +12,22 @@ Author: Remi Delmas, delmasrd@amazon.com
 #ifndef CPROVER_GOTO_INSTRUMENT_CONTRACTS_DYNAMIC_FRAMES_DFCC_INSTRUMENT_H
 #define CPROVER_GOTO_INSTRUMENT_CONTRACTS_DYNAMIC_FRAMES_DFCC_INSTRUMENT_H
 
+#include <goto-programs/goto_convert_class.h>
+
 #include <util/arith_tools.h>
 #include <util/c_types.h>
+#include <util/graph.h>
 #include <util/message.h>
 #include <util/namespace.h>
 #include <util/std_expr.h>
 #include <util/std_types.h>
 
 #include <goto-programs/goto_program.h>
 
+#include <analyses/local_may_alias.h>
+#include <analyses/natural_loops.h>
+#include <goto-instrument/loop_utils.h>
```

Comment:
could we move the write set inference functions to a separate class e.g. `dfcc_instrument_loops.h`